app/controllers/datasets_controller.rb

```diff
@@ -59,21 +68,75 @@ def resubmit
     redirect_to dataset_path(dataset)
   end
 
+  # Outputs the Datacite XML representation of the dataset
+  def datacite
+    @dataset = Dataset.find(params[:id])
+    work = Work.find(@dataset.work_id)
+    resource = Datacite::Resource.new_from_json(work.data_cite)
+    render xml: resource.to_xml
+  end
+
   private
 
     # Only allow a list of trusted parameters through.
     def form_params
-      params.require(:dataset).permit([:title, :work_id, :collection_id])
+      valid_list = [:work_id]
+      params.require(:dataset).permit(valid_list)
     end
 
     def work_params
       {
-        title: form_params[:title],
-        collection_id: form_params[:collection_id]
+        title: params[:title],
+        collection_id: params[:collection_id]
       }
     end
 
     def dataset_params
-      form_params.reject { |x| x.in?(["title", "collection_id"]) }
+      form_params.select { |x| x == "work_id" }
+    end
+
+    def new_creator(given_name, family_name, orcid)
+      return if family_name.blank? && given_name.blank? && orcid.blank?
+      Datacite::Creator.new_person(given_name, family_name, orcid)
+    end
+
+    # Populate the work.data_cite field
+    def update_work_record
+      work = Work.find(form_params[:work_id])
+      work_data = work_params
+      resource = Datacite::Resource.new(title: params["title"])
+
+      # Process the titles
+      resource.titles << Datacite::Title.new(title: params["title_main"])
+      for i in 1..params["existing_title_count"].to_i do
+        if params["title_#{i}"].present?
+          resource.titles << Datacite::Title.new(title: params["title_#{i}"], title_type: params["title_type_#{i}"])
+        end
+      end
+
+      for i in 1..params["new_title_count"].to_i do
+        if params["new_title_#{i}"].present?
+          resource.titles << Datacite::Title.new(title: params["new_title_#{i}"], title_type: params["new_title_type_#{i}"])
+        end
+      end
+
+      # Process the creators
+      for i in 1..params["existing_creator_count"].to_i do
+        creator = new_creator(params["given_name_#{i}"], params["family_name_#{i}"], params["orcid_#{i}"])
+        resource.creators << creator unless creator.nil?
+      end
+
+      for i in 1..params["new_creator_count"].to_i do
+        creator = new_creator(params["new_given_name_#{i}"], params["new_family_name_#{i}"], params["new_orcid_#{i}"])
+        resource.creators << creator unless creator.nil?
+      end
+
+      work_data[:data_cite] = resource.to_json
+      work.update(work_data)
+      work.save!
```

Comment:
This is really the important part of this PR. It process the data from the form (including multiple titles and multiple authors), pushes the data to a Datacite::Resource object. This resource is serialized to JSON and saved to Postgress.